controllers/windowsmachine_controller.go

```diff
@@ -338,19 +340,10 @@ func (r *WindowsMachineReconciler) Reconcile(ctx context.Context, request ctrl.R
 	}
 
 	// Get the IP address associated with the Windows machine, if not error out to requeue again
-	if len(machine.Status.Addresses) == 0 {
-		return ctrl.Result{}, errors.Errorf("machine %s doesn't have any ip addresses defined",
-			machine.Name)
-	}
-	ipAddress := ""
-	for _, address := range machine.Status.Addresses {
-		if address.Type == core.NodeInternalIP {
-			ipAddress = address.Address
-		}
-	}
-	if len(ipAddress) == 0 {
-		return ctrl.Result{}, errors.Errorf("no internal ip address associated with machine %s",
-			machine.Name)
+	ipAddress, err := getInternalIPAddress(machine.Status.Addresses)
+	if err != nil {
+		r.log.V(1).Info("invalid machine", "error", err)
+		return ctrl.Result{}, err
```

Comment:
This is going to result in a debug logging of the error + context, as well as an info logging of just the error.
It makes more sense to me to just return
```suggestion
		return ctrl.Result{}, errors.Wrap(err, "invalid machine")
```

Reply:
Done except I added the Machine name to the returned error.